src/charm.py

```diff
@@ -125,6 +134,9 @@ def func_with_catch_errors(self: "GithubRunnerCharm", event: EventT) -> None:
             self.unit.status = BlockedStatus(
                 "Unauthorized OpenStack connection. Check credentials."
             )
+        except MissingIntegrationDataError as err:
```

Comment:
Rather than raise > catch > waitingstatus, would it be worth differentiating the status of the integration data by None?
What I've done for image-builder relation is that if the integration data is not yet ready, I'll return None values for the data fields. If the integration itself is missing, I'll return a None instead of the data struct.
The drawback of this implementation is that the type hints will include None in the fields:
```
class Dataclass:
    field: None | str
```
and it can become cumbersome doing None check for the fields each time even though it is guaranteed not to be None.
I'm thinking whether I should change the image relation implementation to raise like how you've done here.

Reply:
I would prefer to have the exception control flow rather than handling `None` values.

Reply:
I will modify the image-builder relation to do the same, thank you!